### ISSUE TITLE
rpm with ndb fails to build on macOS due to missing "mremap()"

### ISSUE THREAD

**Issue by Conan-Kudo** [rpm-software-management/rpm#334]
When I attempt to build rpm 4.14 for macOS with the ndb rpmdb backend enabled, it fails to compile due to the Linux-specific [`mremap(2)`](http://man7.org/linux/man-pages/man2/mremap.2.html) syscall missing.

```
depbase=`echo backend/ndb/rpmxdb.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
	/bin/sh ../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H   -I.. -I.. -I../include/  -I../misc -DLOCALEDIR="\"/usr/local/share/locale\"" -DSYSCONFDIR="\"/usr/local/etc\"" -DLOCALSTATEDIR="\"/usr/local/var\"" -DLIBRPMALIAS_FILENAME="\"rpmpopt-4.14.0\"" -DLIBRPMALIAS_EXECPATH="\"/usr/local/bin\"" -I/usr/local/include  -I/usr/local/include -fPIC -DPIC -D_REENTRANT -Wall -Wpointer-arith -Wmissing-prototypes -Wstrict-prototypes  -fno-strict-aliasing -fstack-protector -Wempty-body -DLUA_COMPAT_ALL -DLUA_COMPAT_MODULE -MT backend/ndb/rpmxdb.lo -MD -MP -MF $depbase.Tpo -c -o backend/ndb/rpmxdb.lo backend/ndb/rpmxdb.c &&\
	mv -f $depbase.Tpo $depbase.Plo
libtool: compile:  gcc -DHAVE_CONFIG_H -I.. -I.. -I../include/ -I../misc -DLOCALEDIR=\"/usr/local/share/locale\" -DSYSCONFDIR=\"/usr/local/etc\" -DLOCALSTATEDIR=\"/usr/local/var\" -DLIBRPMALIAS_FILENAME=\"rpmpopt-4.14.0\" -DLIBRPMALIAS_EXECPATH=\"/usr/local/bin\" -I/usr/local/include -I/usr/local/include -fPIC -DPIC -D_REENTRANT -Wall -Wpointer-arith -Wmissing-prototypes -Wstrict-prototypes -fno-strict-aliasing -fstack-protector -Wempty-body -DLUA_COMPAT_ALL -DLUA_COMPAT_MODULE -MT backend/ndb/rpmxdb.lo -MD -MP -MF backend/ndb/.deps/rpmxdb.Tpo -c backend/ndb/rpmxdb.c  -fno-common -DPIC -o backend/ndb/.libs/rpmxdb.o
backend/ndb/rpmxdb.c:168:11: warning: implicit declaration of function 'mremap' is invalid in C99 [-Wimplicit-function-declaration]
        mapped = mremap(slot->mapped - shift, oldsize, newsize, MREMAP_MAYMOVE);
                 ^
backend/ndb/rpmxdb.c:168:58: error: use of undeclared identifier 'MREMAP_MAYMOVE'
        mapped = mremap(slot->mapped - shift, oldsize, newsize, MREMAP_MAYMOVE);
                                                                ^
backend/ndb/rpmxdb.c:764:65: error: use of undeclared identifier 'MREMAP_MAYMOVE'
    newaddr = mremap(xdb->mapped, xdb->mappedlen, newmappedlen, MREMAP_MAYMOVE);
                                                                ^
1 warning and 2 errors generated.
make[3]: *** [backend/ndb/rpmxdb.lo] Error 1
```

**Comment by pmatilai** [comment]
Fixed in commit 8c4bfd74b568d9be138e8ec4fd38dc2ca427d433.

**Comment by pmatilai** [comment]
Dunno but seriously doubt it. Validated patches welcome ;)
